app/json_schemas/json_api_schema.rb

```diff
@@ -99,6 +101,59 @@ def data_properties(require_all: false)
       }
     end
 
+    # Example
+    # extra_fields: :my_method
+    # => extra_fields_result = my_method
+    # => attributes = attributes.merge(extra_fields_result)
+    #
+    # extra_fields: {name: :my_method, required: true, opts: {method_opt: true}}
+    # => extra_fields_result = my_method({method_opt: true})
+    # => attributes = attributes.merge(extra_fields_result)
+    # => required += extra_fields_result.keys
+    #
+    # extra_fields: [:my_method, :another_method]
+    # => extra_fields_result = my_method.merge(another_method)
+    # => attributes = attribtues.merge(extra_fields_result)
+    #
+    # To test use eg::
+    # => MySchema.collection(..., extra_fields: ...)
+    #     .dig(:properties, :data, :items, :properties, :attributes)
+    def get_extra_fields(extra_fields)
+      case extra_fields
+      when Symbol
+        public_send(extra_fields)
+      when Hash
+        extra_fields[:opts] &&
+          public_send(extra_fields[:name], extra_fields[:opts]) || public_send(extra_fields[:name])
```

Comment:
I wonder if we can use double-splat to simplify this:
```suggestion
        public_send(extra_fields[:name], **extra_fields[:opts])
```

This deconstructs the opts, but I think would have the same end result. It has the benefit of handling blank state so that you don't need to.

Reply:
fixed